### PR TITLE
Adding session ticket support to MbedTLS

### PR DESCRIPTION
**Pull request by allanpark**
Adding session ticket support to MbedTLS. 
TLS interface has changed, examples are not updated accordingly!
examples/http-restful-server updated to support MbedTLS if make called with TLS=mbedtls

### CODE REVIEW

File: src/tls_dummy.c [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #include "tls.h"
 
 #if !MG_ENABLE_MBEDTLS && !MG_ENABLE_OPENSSL && !MG_ENABLE_CUSTOM_TLS
-void mg_tls_init(struct mg_connection *c, const struct mg_tls_opts *opts) {
+void* mg_tls_ctx_init(const struct mg_tls_opts *opts) {
+  (void) opts;
+  return NULL;
+}
```
Comment (scaprile) [inline src/tls_dummy.c:7]: This should let the user build with no TLS without complaining

File: examples/http-restful-server/main.c [hunk 2]
```diff
@@ -56,11 +53,19 @@ static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
...
   struct mg_mgr mgr;                            // Event manager
+  struct mg_tls_opts opts = {
+      //.ca = "ca.pem",         // Uncomment to enable two-way SSL
+      .cert = "server.pem",     // Certificate PEM file
+      .certkey = "server.pem",  // This pem contains both cert and key
+  };
   mg_log_set(MG_LL_DEBUG);                      // Set log level
   mg_mgr_init(&mgr);                            // Initialise event manager
+  mgr.tls_ctx = mg_tls_ctx_init(&opts);         // Create TLS context
+
```
Comment (scaprile) [inline examples/http-restful-server/main.c:64]: The example must build and run without TLS

File: examples/http-restful-server/main.c [hunk 3]
```diff
@@ -59,8 +60,8 @@ int main(void) {
   };
   mg_log_set(MG_LL_DEBUG);                      // Set log level
   mg_mgr_init(&mgr);                            // Initialise event manager
-  if(!(mgr.tls_ctx = mg_tls_ctx_init(&opts)))         // Create TLS context
-    return 1;
+  mgr.tls_ctx = mg_tls_ctx_init(&opts);         // Create TLS context
+
```
Comment (scaprile) [inline examples/http-restful-server/main.c]: The example must build and run without TLS

File: examples/http-restful-server/main.c [hunk 4]
```diff
@@ -59,8 +60,8 @@ int main(void) {
   };
   mg_log_set(MG_LL_DEBUG);                      // Set log level
   mg_mgr_init(&mgr);                            // Initialise event manager
-  if(!(mgr.tls_ctx = mg_tls_ctx_init(&opts)))         // Create TLS context
-    return 1;
```
Comment (scaprile) [inline examples/http-restful-server/main.c]: The example must build and run without TLS

File: examples/http-restful-server/main.c [hunk 5]
```diff
@@ -15,14 +15,15 @@
...
 static const char *s_http_addr = "http://0.0.0.0:8000";    // HTTP port
-static const char *s_https_addr = "https://0.0.0.0:443";  // HTTPS port
+static const char *s_https_addr = "https://0.0.0.0:8443";  // HTTPS port
 static const char *s_root_dir = ".";
 
 // We use the same event handler function for HTTP and HTTPS connections
 // fn_data is NULL for plain HTTP, and non-NULL for HTTPS
 static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
   if (ev == MG_EV_ACCEPT && fn_data != NULL) {
-    struct mg_tls_session_opts opts = {0};
```
Comment (scaprile) [inline examples/http-restful-server/main.c]: We've had some problems with this in some compilers

File: src/tls_dummy.c [hunk 6]
```diff
@@ -3,7 +3,6 @@
 #if !MG_ENABLE_MBEDTLS && !MG_ENABLE_OPENSSL && !MG_ENABLE_CUSTOM_TLS
 void* mg_tls_ctx_init(const struct mg_tls_opts *opts) {
   (void) opts;
-  MG_ERROR(("TLS is not enabled"));
```
Comment (scaprile) [inline src/tls_dummy.c]: This should let the user build and run with no TLS, using the same code base, without complaining

**Comment by scaprile** [comment]
Unit tests now build with MBEDTLS but some fail. I'll look at it if needed, just let me know, I'm not fluent in this.
Unit tests don't build with OPENSSL, this change breaks the interface.
Changed where I left those comments regarding the example,
Once unit tests pass, I'll make the examples tests pass and then merge.

**Comment by cpq** [comment]
I think tickets support is already in the master branch - can we close this?

**Pull request closed without merging**